ocaml/xcp-rrdd/scripts/rrdd/rrdd.py

```diff
@@ -310,19 +311,25 @@ def update(self):
         metadata_json = json.dumps(metadata, sort_keys=True).encode('utf-8')
         metadata_checksum = crc32(metadata_json) & 0xffffffff
 
-        self.dest.seek(0)
-        self.dest.write('DATASOURCES'.encode())
-        self.dest.write(pack(">LLLQ",
-                             data_checksum,
-                             metadata_checksum,
-                             len(self.datasources),
-                             timestamp))
+        # First write the updated data and metadata
+        encoded_datasource_header = 'DATASOURCES'.encode()
+        # DATASOURCES + 20 for 32 + 32 + 32 + 64
+        self.dest.seek(len(encoded_datasource_header) + 20)
         for val in data_values:
             # This is already big endian encoded
             self.dest.write(val)
 
         self.dest.write(pack(">L", len(metadata_json)))
         self.dest.write(metadata_json)
+
```

Comment:
do we need to `self.dest.flush()` here or does seek imply a flush?

Reply:
probably not a bad idea although there is a flush after all the writes are complete